worlds/pokemon_emerald/__init__.py

```diff
@@ -233,10 +234,27 @@ def generate_early(self) -> None:
                 max_norman_count = 4
 
         if self.options.norman_count.value > max_norman_count:
-            logging.warning("Pokemon Emerald: Norman requirements for Player %s (%s) are unsafe in combination with "
+            logging.warning("Pokemon Emerald: Norman requirements for player %s (%s) are unsafe in combination with "
                             "other settings. Reducing to 4.", self.player, self.player_name)
             self.options.norman_count.value = max_norman_count
 
+        # Prevent player from setting badges/hms to non-local if shuffling them
+        if self.options.badges == RandomizeBadges.option_shuffle:
+            for item in self.item_name_groups["Badge"]:
+                if item in self.options.non_local_items.value:
+                    logging.warning("Pokemon Emerald: Player %s (%s) set badges to be shuffled in their world, but "
+                                    "then added %s to non-local items. Removing the item from non-local items.",
+                                    self.player, self.player_name, item)
+                    self.options.non_local_items.value.remove(item)
+
+        if self.options.hms == RandomizeHms.option_shuffle:
+            for item in self.item_name_groups["HM"]:
+                if item in self.options.non_local_items.value:
+                    logging.warning("Pokemon Emerald: Player %s (%s) set hms to be shuffled in their world, but "
+                                    "then added %s to non-local items. Removing the item from non-local items.",
+                                    self.player, self.player_name, item)
+                    self.options.non_local_items.value.remove(item)
```

Comment:
I think that choosing Badge/HM shuffle implies that the Badges/HMs have been chosen to be local, so I think it is better to add the items to local_items instead.

Local items have priority over non-local items.
```suggestion
        # Shuffled badges/hms will always be placed locally, so add them to local_items
        if self.options.badges == RandomizeBadges.option_shuffle:
            self.options.local_items.value.update(self.item_name_groups["Badge"])

        if self.options.hms == RandomizeHms.option_shuffle:
            self.options.local_items.value.update(self.item_name_groups["HM"])
```

Reply:
I think this seems fine _as long as_ this behavior of local items overriding nonlocal items is canonical somehow. I did see that this is how it behaves in the code, but it feels like a sort of arbitrary decision. Whatever core dev(s) review can give their opinion, I suppose.